grammar.js

```diff
@@ -2269,11 +2271,20 @@ module.exports = grammar({
           $.keyword_references,
           $.object_reference,
           $.ordered_columns,
-          optional(
+          repeat(
             seq(
               $.keyword_on,
-              $.keyword_delete,
-              $.keyword_cascade,
+              choice($.keyword_delete, $.keyword_update),
+              choice(
+                seq($.keyword_no, $.keyword_action),
+                $.keyword_restrict,
+                $.keyword_cascade,
+                seq(
+                  $.keyword_set,
+                  choice($.keyword_null, $.keyword_default),
```

Comment:
I think `set null` and `set default` (this one is missing), can optionally be followed by a column name. In our parser the column name is represented by the `object_reference` node. (`paren_list()` of `object_reference`

https://www.postgresql.org/docs/current/sql-altertable.html (under `referential_action`)

Reply:
You cannot put can just put an object_reference just an identifier:
```
toto=# create table test1(id integer references test0(id) on delete set null(test.id));
ERROR:  syntax error at or near "."
LINE 1: ...d integer references test0(id) on delete set null(test.id));
```

Reply:
Also line 2273 `$.ordered_columns` should be `paren_list($.identifier)` at least on postgres.

Reply:
This worked for me on postgres 15.3

```sql
CREATE TABLE table1 (
    id serial PRIMARY KEY,
    column_name1 integer,
    column_name2 integer
);

CREATE TABLE table2 (
    id serial PRIMARY KEY,
    table1_id integer REFERENCES table1(id) ON DELETE CASCADE
);

ALTER TABLE table1
DROP CONSTRAINT IF EXISTS table1_column_name1_fkey;

ALTER TABLE table1
ADD CONSTRAINT table1_column_name1_fkey
FOREIGN KEY (column_name1)
REFERENCES table2(id)
ON DELETE SET NULL (column_name1);
```

Reply:
You are right about `object_reference`. This won't work. `identifier` does work however.